internal/db/mailbox.go

```diff
@@ -70,27 +70,41 @@ func RenameMailboxWithRemoteID(ctx context.Context, tx *ent.Tx, mboxID imap.Labe
 	return nil
 }
 
-func DeleteMailboxWithRemoteID(ctx context.Context, tx *ent.Tx, mboxID imap.LabelID) error {
-	// get uid validity
+func DeleteMailboxWithRemoteID(ctx context.Context, tx *ent.Tx, mboxID imap.LabelID) (imap.UID, bool, error) {
 	mbox, err := tx.Mailbox.Query().
 		Where(mailbox.RemoteID(mboxID)).
 		Select(mailbox.FieldUIDValidity).
 		Only(ctx)
 	if err != nil {
-		return err
+		return 0, false, err
 	}
 
-	if err := updateGlobalUIDValidity(ctx, tx, mbox.UIDValidity); err != nil {
-		return err
+	curUIDValidity, err := getGlobalUIDValidity(ctx, tx)
+	if err != nil {
+		return 0, false, err
+	}
+
+	var newUIDValidity imap.UID
+
+	if mbox.UIDValidity == curUIDValidity {
+		newUIDValidity = curUIDValidity.Add(1)
+	} else {
+		newUIDValidity = curUIDValidity
+	}
+
+	if newUIDValidity > curUIDValidity {
+		if err := setGlobalUIDValidity(ctx, tx, newUIDValidity); err != nil {
+			return 0, false, err
+		}
 	}
 
 	if _, err := tx.Mailbox.Delete().
 		Where(mailbox.RemoteID(mboxID)).
 		Exec(ctx); err != nil {
-		return err
+		return 0, false, err
 	}
 
-	return nil
+	return newUIDValidity, newUIDValidity > curUIDValidity, nil
 }
 
 func UpdateRemoteMailboxID(ctx context.Context, tx *ent.Tx, internalID imap.InternalMailboxID, remoteID imap.LabelID) error {
```

Comment:
Is this function unused now?

Reply:
It is still used and is not related to these changes. Are you referring to a different one?

Reply:
Sorry, my mistake.